### ISSUE TITLE
Fix display logic (or the dataset itself) for overlapping Cliopatria shapes

### ISSUE THREAD

**Issue by edwardchalstrey1** [Seshat-Global-History-Databank/seshat#36]
This is the logic of which shapes to display, based on whether the display dropdown is set to full polities or components:

```
function shouldDisplayComponent(displayComponent, shape) {
    if (displayComponent == 'polities'
        && (shape.member_of === null || shape.member_of === '')) {
        return true;
    } else if (displayComponent == 'components'
        && (shape.components === null || shape.components === '')) {
        return true;
    } else {
        return false;
    }
}
```

In plain English:

```
If the dropdown is set to polities:
    Show shapes that are not a member of anything (are highest level composites)
If the dropdown is set to components:
    Show shapes that don't contain any components (are lowest level components)
```

You can see here: http://51.141.239.61:8000/core/world_map if you navigate to the year 1519 (with the dropdown set to "Full polities") and you'll see an example of a shape that currently displays when it shouldn't, which is "Kingdom of Spain" (try clicking on Spain and Habsburg monarchy areas) that is actually called "(Kingdom of Spain)" in the GeoJSON:

```
"Name": "(Kingdom of Spain)", "FromYear": 1519, "ToYear": 1520, "Area": 943945.48080229305, "Type": "POLITY", "Wikipedia": "Spain", "SeshatID": "", "MemberOf": "", "Components": "Kingdom of Spain"
```

^ This should be changed to include `"MemberOf": "(Personal union of Spanish Empire with Habsburg Monarchy)"` so it is not displayed

So the composite shape it currently overlaps with is:

```
"Name": "(Personal union of Spanish Empire with Habsburg Monarchy)", "FromYear": 1519, "ToYear": 1520, "Area": 1064933.8920869608, "Type": "POLITY", "Wikipedia": "Charles V, Holy Roman Emperor", "SeshatID": "es_spanish_emp_1;at_habsburg_1", "MemberOf": "", "Components": "Kingdom of Spain;(Spanish Empire);Habsburg Monarchy;(Habsburg Monarchy)"
```

**Comment by jbennettgit** [comment]
Good eye. Yep looks like a bug in the dataset due to editing code interactions.  I'll look at a fix.  Could be next week depending on how much time I get during 'vacation' here.

**Comment by jbennettgit** [comment]
Fixed the bug when computing missing member of and components when there are multiple levels of associated polities; new data set available.  Just a comment about about the logic above (now that the data bug is fixed).  I think most of the time the code above should work with all the current examples of a deep multi-level hierarchy of associated polities but that is because often the components have the same names as one of the intermediate associated polities so you can't tell the difference.  This will be the case with the personal union example that triggered this bug: the (Spanish Empire) and (Kingdom of Spain) intermediates have the Kingdom of Spain component, similarly the (Habsburg Monarchy) has Habsburg Monarchy.  If we ever get a skew in the names and want to show an intermediate this code won't work.  But let's wait for an example.

**Comment by edwardchalstrey1** [comment]
> Fixed the bug when computing missing member of and components when there are multiple levels of associated polities; new data set available. Just a comment about about the logic above (now that the data bug is fixed). I think most of the time the code above should work with all the current examples of a deep multi-level hierarchy of associated polities but that is because often the components have the same names as one of the intermediate associated polities so you can't tell the difference. This will be the case with the personal union example that triggered this bug: the (Spanish Empire) and (Kingdom of Spain) intermediates have the Kingdom of Spain component, similarly the (Habsburg Monarchy) has Habsburg Monarchy. If we ever get a skew in the names and want to show an intermediate this code won't work. But let's wait for an example.

Brilliant, this is looking good to me now
